### PR TITLE
Rebuild as many projects as possible

### PR DESCRIPTION
**Pull request by RikkiGibson**
This modifies the rebuild leg to rebuild as many bits as possible.

This also introduces some changes to:
1. not fail the rebuild leg when assemblies that don't have compilation options/metadata references *at all* are found.
2. not crash the program when such assemblies are found.
3. embed ildasm with the utility as the Helix machines don't always have it available in the hard coded location we assume it will be found in. This makes it so we can actually get a debug artifact when the rebuild fails in CI.

Results of rebuilding *everything* in Roslyn repo are detailed here: https://gist.github.com/RikkiGibson/77cad2adb2e35552aa2008834d6f69a5

Please take a look @tmat @jaredpar @cston

### CODE REVIEW

**Comment by RikkiGibson** [comment]
bleh, the helix machines don't have ildasm in the conventional location. will probably want to look at embedding ildasm or switching queues back to AzDO vmImages.

File: eng/Versions.props [hunk 1]
```diff
@@ -32,7 +32,8 @@
     <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.20623.3</MicrosoftCodeAnalysisTestingVersion>
     <CodeStyleAnalyzerVersion>3.8.0</CodeStyleAnalyzerVersion>
     <VisualStudioEditorPackagesVersion>16.8.181</VisualStudioEditorPackagesVersion>
-    <ILToolsPackageVersion>5.0.0-alpha1.19409.1</ILToolsPackageVersion>
+    <ILAsmPackageVersion>5.0.0-alpha1.19409.1</ILAsmPackageVersion>
+    <ILDAsmPackageVersion>5.0.0-preview.1.20112.8</ILDAsmPackageVersion>
```
Comment (jaredpar) [inline eng/Versions.props:36]: Why are we using different versions here?

**Review by jaredpar — APPROVED** [review]

File: eng/Versions.props (hunk 1 above)
Comment (RikkiGibson) [reply eng/Versions.props:36]: When I tried using the same, original version for both, NuGet complained me that it couldn’t find such a version of ildasm and chose the version that you see instead.

File: src/Tools/BuildValidator/IldasmUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Roslyn.Utilities;
+
+namespace BuildValidator
+{
+    public static class IldasmUtilities
```
Comment (cston) [inline src/Tools/BuildValidator/IldasmUtilities.cs:13]: >public [](start = 4, length = 6)

`internal`?

File: eng/test-rebuild.ps1 [hunk 3]
```diff
@@ -36,13 +36,45 @@ try {
...
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.Remote.Workspaces`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.ResultProvider`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.Scripting`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.Scripting.TestUtilities`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.TestSourceGenerator`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/Microsoft.CodeAnalysis.Workspaces`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/PrepareTests`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/RunTests`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/vbc/$configuration/netcoreapp3.1`"" +
+  " --assembliesPath `"$ArtifactsDir/obj/VBCSCompiler/$configuration/netcoreapp3.1`"" +
```
Comment (jaredpar) [inline eng/test-rebuild.ps1:72]: image

**Review by jaredpar — APPROVED** [review]

**Pull request merged**